test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -294,6 +303,27 @@ class ExtAuthzHttpIntegrationTest : public HttpIntegrationTest,
     initiateClientConnection();
     waitForExtAuthzRequest();
 
+    AssertionResult result =
+        fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_);
+    RELEASE_ASSERT(result, result.message());
+    result = upstream_request_->waitForEndStream(*dispatcher_);
+    RELEASE_ASSERT(result, result.message());
+
+    // The original client request header value of "baz" is "foo". Since we configure to "override"
+    // the value of "baz", we expect the request headers to be sent to upstream contain only one
+    // "baz" with value "baz" (set by the authorization server).
+    EXPECT_THAT(upstream_request_->headers(), Http::HeaderValueOf("baz", "baz"));
+
+    // The original client request header value of "bat" is "foo". Since we configure to "append"
+    // the value of "bat", we expect the request headers to be sent to upstream contain two "bat"s,
+    // with values: "foo" and "bar" (the "bat: bar" header is appended by the authorization server).
+    const auto& baz_values =
+        TestUtility::getAllHeadersValuesForKey(upstream_request_->headers(), "bat");
```

Comment:
Sorry, let's use `Http::IsSubsetOfHeaders` instead.

```c++
EXPECT_THAT(upstream_request_->headers(), Not(IsSubsetOfHeaders(TestRequestHeaderMapImpl{{"bat", "foo"}, {"bat", "bar"}})));
```

Reply:
It seems I cannot pass the test by adding this, errors like:
```
test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:324: Failure
Value of: upstream_request_->headers()
Expected: is a subset of headers:
'bat', 'foo'
'bat', 'bar'

  Actual: 16-byte object <C0-2D 12-05 00-00 00-00 F8-39 12-05 00-00 00-00> (of type Envoy::Http::RequestHeaderMap), where the following elements don't match any matchers:
element #0: (":authority", "host"),
element #1: (":path", "/"),
element #2: (":method", "GET"),
element #3: ("x-case-sensitive-header", "Case-Sensitive"),
element #4: ("baz", "baz"),
element #6: ("content-length", "0"),
element #7: ("x-forwarded-proto", "http"),
element #8: ("x-request-id", "ecee8a3a-4812-4706-87bc-96376c788078"),
element #10: ("x-envoy-expected-rq-timeout-ms", "15000")
Stack trace:
  0x2212a19: Envoy::ExtAuthzHttpIntegrationTest::setupWithDisabledCaseSensitiveStringMatcher()
  0x220fb3e: Envoy::ExtAuthzHttpIntegrationTest_DefaultCaseSensitiveStringMatcher_Test::TestBody()
  0x5021fb4: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x501288b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x50002e3: testing::Test::Run()
  0x5000ca7: testing::TestInfo::Run()

```

Reply:
Exactly, that was a wrong comment, you should switch the position 🙂 (I think it is obvious from the error message).

```c++
EXPECT_THAT(Http::TestRequestHeaderMapImpl{{"bat", "foo"}, {"bat", "bar"}}), Http::IsSubsetOfHeaders(upstream_request_->headers());
```